src/components/CoreTabs.vue

```diff
@@ -152,20 +154,36 @@ import { mapGetters } from 'vuex'
         this.activeTab = tab
       },
       close(tab) {
-
         if (this.activeTab === tab) {
           if(tab === this.lastTab) {
             this.previousTab()
           } else {
             this.nextTab()
           }
         }
+
         this.tabItems = _.without(this.tabItems, tab)
         if (tab.query && tab.query.id) {
           tab.query.reload()
         }
       },
+      closeAll() {
+        this.tabItems = []
+      },
+      closeOther(tab) {
+        this.tabItems = [tab]
+        this.activeTab = tab;
+        if (tab.query && tab.query.id) {
+          tab.query.reload()
+        }
+      },
+      duplicate(tab) {
+        let duplicatedTab = Object.assign( Object.create( Object.getPrototypeOf(tab)), tab)
```

Comment:
So this cloning doesn't work here because table tabs have a `table` or a `query` embedded in them which is the same table object as comes from the store.

So think we have to manually set id/title/querytext/etc

Some things this needs to do:
1. Manually set query text on a new query object so that saved queries can be duplicated (only one copy of a saved query should be open at a time)
2. Don't clone, but just use the same `table` object for new table tabs, so that we maintain referential integrity with the store

Other questions:

Reply:
I commit a new version of the duplicate function, which builds a new object instead of cloning the old one.
Please check if that version is ok.